cbt_export_changes.py

```diff
@@ -24,37 +24,36 @@
 from bitstring import BitStream
 import argparse
 import os
+import re
 
 # CBT tracks 64KB blocks. Therefore each bit in the bitmap corresponds to a
 # 64KB block on the VDI.
 changed_block_size = 64 * 1024
 certfile = "cacert.pem"
 
 
-def collect_host_certificates(session):
-    # Note that this function is only necessary for self-signed certificates
-    # For certificates signed by a CA you will need to collect the public
-    # certificates from the CA.
-    host_list = session.xenapi.host.get_all()
-    with open(certfile, 'w') as cert_out:
-        for host in host_list:
-            cert_out.write(session.xenapi.host.get_server_certificate(host)
-                           + "\n")
+def find_fqdn(cert_text):
+    from cryptography import x509
+    from cryptography.hazmat.backends import default_backend
+    cert = x509.load_pem_x509_certificate(cert_text, default_backend())
+    cert_subject = str(cert.subject)
+    cert_hostname = re.search(r'value=u\'(.*)\'', cert_subject)
+    return cert_hostname.group(1)
 
 
 def delete_host_certificates_file():
     os.remove(certfile)
 
 
-def get_changed_blocks(host, export_name, bitmap, bitmap_output_path):
+def get_changed_blocks(host, export_name, tls_hostname, bitmap, bitmap_output_path):

```

Comment:
`tls_hostname` would be better named `tls_subject` because it doesn't need to correspond to a hostname.
But see other comment about using the certificate (or its public key) instead.

Reply:
I used tls_hostname to match the terminology from the linux nbd client. Seeing as thats the other example we will point partners at I think its useful to have consistent terminology.